Extension/src/LanguageServer/ui_new.ts

```diff
@@ -0,0 +1,528 @@
+    }
+
+    private setCodeAnalysisTotal(total: number): void {
+        if (!this.isRunningCodeAnalysis) {
+            return; // Occurs when a multi-root workspace is activated.
+        }
+        this.codeAnalysisTotal = total;
+        this.updateCodeAnalysisTooltip();
+    }
+
+    private get ReferencesCommand(): ReferencesCommandMode {
+        return this.referencesStatusBarItem.tooltip === "" ? ReferencesCommandMode.None :
+            (this.referencesStatusBarItem.tooltip === referencesCommandModeToString(ReferencesCommandMode.Find) ? ReferencesCommandMode.Find :
+                (this.referencesStatusBarItem.tooltip === referencesCommandModeToString(ReferencesCommandMode.Rename) ? ReferencesCommandMode.Rename :
+                    ReferencesCommandMode.Peek));
+    }
+
+    private set ReferencesCommand(val: ReferencesCommandMode) {
+        if (val === ReferencesCommandMode.None) {
+            this.referencesStatusBarItem.text = "";
+            this.ShowReferencesIcon = false;
+        } else {
+            this.referencesStatusBarItem.text = "$(search)";
+            this.referencesStatusBarItem.tooltip =  referencesCommandModeToString(val) + (val !== ReferencesCommandMode.Find ? "" : this.referencesPreviewTooltip);
+            this.ShowReferencesIcon = true;
+        }
+    }
+
+    private set ShowReferencesIcon(show: boolean) {
+        if (show && this.ReferencesCommand !== ReferencesCommandMode.None) {
+            this.referencesStatusBarItem.show();
+        } else {
+            this.referencesStatusBarItem.hide();
+        }
+    }
+
+    public bind(client: Client): void {
+        client.ParsingWorkspaceChanged(value => { this.setIsParsingWorkspace(value); });
+        client.ParsingWorkspacePausableChanged(value => { this.setIsParsingWorkspacePausable(value); });
+        client.ParsingWorkspacePausedChanged(value => { this.setIsParsingWorkspacePaused(value); });
+        client.ParsingFilesChanged(value => { this.setIsParsingFiles(value); });
+        client.IntelliSenseParsingChanged(value => { this.setIsUpdatingIntelliSense(value); });
+        client.RunningCodeAnalysisChanged(value => { this.setIsRunningCodeAnalysis(value); });
+        client.CodeAnalysisPausedChanged(value => { this.setIsCodeAnalysisPaused(value); });
+        client.CodeAnalysisProcessedChanged(value => { this.setCodeAnalysisProcessed(value); });
+        client.CodeAnalysisTotalChanged(value => { this.setCodeAnalysisTotal(value); });
+        client.ReferencesCommandModeChanged(value => { this.ReferencesCommand = value; });
+        client.TagParserStatusChanged(value => { this.TagParseStatus = value; });
+        client.ActiveConfigChanged(value => { this.ActiveConfig = value; });
+    }
+
+    public async showConfigurations(configurationNames: string[]): Promise<number> {
+        const options: vscode.QuickPickOptions = {};
+        options.placeHolder = localize("select.a.configuration", "Select a Configuration...");
```

Comment:
Why is the hover "Select a Configuration" when the link shows "Edit Configuration"? I mentioned this earlier, but I'm not sure what the answer was.

Reply:
I don't remember too. But it was based off the spec so I believe it had to do with some agreement from that